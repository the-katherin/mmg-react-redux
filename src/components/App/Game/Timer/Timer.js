import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getIsFinished } from "../../../../selectors";
import { getTime } from '../../../../actions';

export class Timer extends Component {

    constructor(props) {
        super(props);
        this.state = { time: 0 };

    }

    componentDidMount() {
        this.timerID = setInterval(
            () => this.tick(),
            1000
        );
    }

    componentWillUnmount() {

        if (this.props.isFinished) {
            this.props.dispatch(getTime(this.state.time));
        }

        clearInterval(this.timerID);
    }

    tick() {
        this.setState((prevState) => ({
            time: prevState.time + 1
        }));
    }

    render() {
        return (
            <div className="timer">
                <h3> Time: {this.state.time}</h3>
            </div>
        );
    }
};


Timer.propTypes = {
    isFinished: PropTypes.bool.isRequired,
};

export default connect(
    state => ({
        isFinished: getIsFinished(state),
    }),

)(Timer);
