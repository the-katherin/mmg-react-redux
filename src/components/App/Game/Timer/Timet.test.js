import React from "react";
import { Timer } from "./Timer";

let wrapper;

describe("Timer", () => {
    beforeEach(() => {
        wrapper = shallow(<Timer
            isFinished={false}
        />)
    });

    test("match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });


});
