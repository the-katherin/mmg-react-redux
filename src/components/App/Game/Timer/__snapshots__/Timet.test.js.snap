// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer match snapshot 1`] = `
<div
  className="timer"
>
  <h3>
     Time: 
    0
  </h3>
</div>
`;
